### PR TITLE
Airflow annotation for task ID

### LINKED ISSUE #1608: ArmadaOperator: Add airflow task id to Armada annotations
When a user submits an Airflow task, we should add the airflow task id to the Armada annotations.

An example can be below: 
```
jobSetId: test
priority: 100
namespace: batch-test
annotations:
  armadaproject.io/taskRunId: 0a016adf-8325-1d6a-8183-268de7a625ea 
  ````

### PR DESCRIPTION
**Pull request by suprjinx**
Closes #1608

### CODE REVIEW

File: third_party/airflow/armada/operators/armada.py [hunk 1]
```diff
@@ -102,3 +104,23 @@ def execute(self, context) -> None:
...
+    :param job_request_items: The job request items we plan to send to armada
+
+    :return: annotated job request items for armada
+    """
+    task_instance = context["ti"]
+    task_id = task_instance.task_id
+
+    for item_list in job_request_items:
+        for item in item_list:
+            item.annotations["armadaproject.io/taskId"] = task_id
```
Comment (ClifHouck) [inline third_party/airflow/armada/operators/armada.py:124]: Should annotations be stored as constants? Is there a spec somewhere for them?

File: third_party/airflow/tests/unit/test_airflow_operator_mock.py [hunk 2]
```diff
@@ -69,65 +71,19 @@ def sleep_job():
         ],
     )
     return [submit_pb2.JobSubmitRequestItem(priority=0, pod_spec=pod)]
-
-
-def test_job_service_health():
```
Comment (ClifHouck) [inline third_party/airflow/tests/unit/test_airflow_operator_mock.py:74]: Why'd a bunch of tests disappear? They seem unrelated to your change.

Comment (suprjinx) [reply third_party/airflow/tests/unit/test_airflow_operator_mock.py:74]: hmm that's not right -- fixing

File: third_party/airflow/armada/operators/armada.py (hunk 1 above)
Comment (suprjinx) [reply third_party/airflow/armada/operators/armada.py]: re: spec, there is only the ticket (#1608) AFAIK. 

re: const -- i usually think that a const is unecessary for singular usage, but I moved this to a const var.

File: third_party/airflow/tests/unit/test_airflow_operator_mock.py (hunk 2 above)
Comment (suprjinx) [reply third_party/airflow/tests/unit/test_airflow_operator_mock.py]: errant deletion, fixed

**Review by ClifHouck — DISMISSED** [review]
LGTM

File: third_party/airflow/armada/operators/armada.py [hunk 3]
```diff
@@ -29,6 +29,8 @@
 
 armada_logger = logging.getLogger("airflow.task")
 
+ANNOTATION_TASK_ID = "armadaproject.io/taskId"
```
Comment (kannon92) [inline third_party/airflow/armada/operators/armada.py:32]: This is actually configurable from lookout so we should look more into this. Env variable or read from lookout confit.

File: third_party/airflow/armada/operators/armada.py [hunk 4]
```diff
@@ -102,3 +106,22 @@ def execute(self, context) -> None:
             "Armada Job finished with %s and message: %s", job_state, job_message
         )
         airflow_error(job_state, self.name, job_id)
+
+
+def annotate_job_request_items(context, job_request_items):
```
Comment (kannon92) [inline third_party/airflow/armada/operators/armada.py:111]: Can you add return types

File: third_party/airflow/armada/operators/armada.py (hunk 3 above)
Comment (kannon92) [reply third_party/airflow/armada/operators/armada.py:32]: No, I mean that lookout will decide what annotations to show up if that match the part before '/'.  

https://github.com/G-Research/armada/blob/master/config/lookout/config.yaml#L16

So annotations that are displayed to user in lookout have to have a form that is present in the config.  Otherwise, we ignore them.  

The pod object has to add the annotation that matches how we deploy lookout to display annotations.  We should probably see if this is really just a constant value or changes depending on production environment in GR.

Comment (suprjinx) [reply third_party/airflow/armada/operators/armada.py]: it doesn't seem like airflow module should pull in lookout config

File: third_party/airflow/armada/operators/armada.py (hunk 4 above)
Comment (suprjinx) [reply third_party/airflow/armada/operators/armada.py]: added

File: localdev/docker-compose.yaml [hunk 5]
```diff
@@ -4,6 +4,9 @@ networks:
   kind:
     external: true
```
Comment (kannon92) [inline localdev/docker-compose.yaml:6]: Can we remove these changes from this PR?

Comment (suprjinx) [reply localdev/docker-compose.yaml:6]: oops, yes will remove -- should be in master now

File: third_party/airflow/armada/operators/armada.py [hunk 6]
```diff
@@ -120,3 +127,43 @@ def _get_lookout_url(self, job_id: str) -> str:
...
+    :param job_request_items: The job request items to be sent to armada
+
+    :return: annotated job request items for armada
+    """
+    task_instance = context["ti"]
+    task_id = task_instance.task_id
+    run_id = context["run_id"]
+    dag_id = context["dag"].dag_id
+
+    for item in job_request_items:
```
Comment (kannon92) [inline third_party/airflow/armada/operators/armada.py:149]: My python is weak.  Does this work if annotations are not defined?  I think they are optional fields?

Comment (suprjinx) [reply third_party/airflow/armada/operators/armada.py:149]: i think it's an empty dictionary when empty and this should work -- would you mind running in your airflow setup to confirm? I have not set that up yet

Comment (kannon92) [reply third_party/airflow/armada/operators/armada.py:149]: Sounds like a perfect time to do that.. ;)

Comment (suprjinx) [reply third_party/airflow/armada/operators/armada.py:149]: the sleep_job() test function provides the default annotations value in the unit test, and this works

**Review by kannon92 — APPROVED** [review]
